File: application/smartgarden_flutter_socket/lib/src/models/node_meta.dart
class NodeMeta {
  final int nodeId;
  final String name;
  final String location;
  final DateTime? lastSeenAt;

  NodeMeta({
    required this.nodeId,
    required this.name,
    required this.location,
    this.lastSeenAt,
  });

  factory NodeMeta.fromJson(Map<String, dynamic> j) => NodeMeta(
        nodeId: (j['node_id'] as num).toInt(),
        name: (j['name'] ?? '') as String,
        location: (j['location'] ?? '') as String,
        lastSeenAt: j['lastSeenAt'] != null ? DateTime.parse(j['lastSeenAt'] as String) : null,
      );
}

File: server/src/models/NodeMeta.js
import mongoose from "mongoose";

const NodeMetaSchema = new mongoose.Schema(
  {
    node_id: { type: Number, unique: true, required: true },
    name: { type: String, default: "" },
    location: { type: String, default: "" },
    lastSeenAt: { type: Date }
  },
  { timestamps: { createdAt: "createdAt", updatedAt: "updatedAt" } }
);

export default mongoose.model("NodeMeta", NodeMetaSchema);
